autoload/lexima/template_rules.vim

```diff
@@ -20,236 +20,331 @@
-            \ 'filetype': c.filetypes
-            \ })
-        call lexima#add_rule({
-            \ 'char': c.char,
-            \ 'at': '\%# '.c.char.'}',
-            \ 'leave': 3,
-            \ 'filetype': c.filetypes
-            \ })
-        call lexima#add_rule({
-            \ 'char': '<Space>',
-            \ 'at': '{'.c.char.'\%#'.c.char.'}',
-            \ 'input': '<Space>',
-            \ 'input_after': '<Space>',
-            \ 'filetype': c.filetypes
-            \ })
-    endfor
-
-    let s:template_tags = [
-        \ {'tag': 'autoescape', 'filetypes': ['jinja', 'htmljinja', 'django', 'htmldjango', 'twig',]},
-        \ {'tag': 'block'     , 'filetypes': ['jinja', 'htmljinja', 'django', 'htmldjango', 'twig',]},
+        \ 'filetype': ['jinja', 'htmljinja', 'django', 'htmldjango', 'liquid', 'twig', 'html.twig',],
+        \ })
+  call lexima#add_rule({
+        \ 'char': '<BS>',
+        \ 'at': '-}}\%#',
+        \ 'input': '<BS><BS><BS><BS>',
+        \ 'input_after': ' -}}',
+        \ 'filetype': ['jinja', 'htmljinja', 'django', 'htmldjango', 'liquid', 'twig', 'html.twig',],
+        \ })
+  call lexima#add_rule({
+        \ 'char': '<BS>',
+        \ 'at': '{{- \%# -}}',
+        \ 'delete': 1,
+        \ 'filetype': ['jinja', 'htmljinja', 'django', 'htmldjango', 'liquid', 'twig', 'html.twig',],
+        \ })
+  call lexima#add_rule({
+        \ 'char': '<BS>',
+        \ 'at': '{{-\%#-}}',
+        \ 'delete': 1,
+        \ 'filetype': ['jinja', 'htmljinja', 'django', 'htmldjango', 'liquid', 'twig', 'html.twig',],
+        \ })
+
+  for c in s:chars
+    call lexima#add_rule({
+          \ 'char': c.char,
+          \ 'at': '{\%#}',
+          \ 'input_after': c.char,
+          \ 'filetype': c.filetypes
+          \ })
+    call lexima#add_rule({
+          \ 'char': '-',
+          \ 'at': '{'.c.char.'\%#'.c.char.'}',
+          \ 'input_after': '-',
+          \ 'filetype': c.filetypes
+          \ })
+    call lexima#add_rule({
+          \ 'char': c.char,
+          \ 'at': '\%#'.c.char.'}',
+          \ 'leave': 2,
+          \ 'filetype': c.filetypes
+          \ })
+    call lexima#add_rule({
+          \ 'char': c.char,
+          \ 'at': '\%#-'.c.char.'}',
+          \ 'leave': 3,
+          \ 'filetype': c.filetypes
+          \ })
+    call lexima#add_rule({
+          \ 'char': '<BS>',
+          \ 'at': '{'.c.char.'\%#'.c.char.'}',
+          \ 'delete': 1,
+          \ 'filetype': c.filetypes
+          \ })
+    call lexima#add_rule({
+          \ 'char': '<BS>',
+          \ 'at': '{'.c.char.'-\%#-'.c.char.'}',
+          \ 'delete': 1,
+          \ 'filetype': c.filetypes
+          \ })
+    call lexima#add_rule({
+          \ 'char': '<BS>',
+          \ 'at': '{'.c.char.' \%# '.c.char.'}',
+          \ 'delete': 1,
+          \ 'filetype': c.filetypes
+          \ })
+    call lexima#add_rule({
+          \ 'char': '<BS>',
+          \ 'at': '{-'.c.char.' \%# -'.c.char.'}',
```

Comment:
I couldn't make the `{%- | -%}` mappings work with `<BS>`, tracked it down to this line. Changing it to this made it work for me: `\ 'at': '{'.c.char.'- \%# -'.c.char.'}',`